src/coreclr/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -333,7 +333,22 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                             genHWIntrinsic_R_R_RM_R(node, ins);
                             break;
                         }
+                        case NI_AVXVNNI_MultiplyWideningAndAdd:
+                        case NI_AVXVNNI_MultiplyWideningAndAddSaturate:
+                        {
+                            assert(targetReg != REG_NA);
+                            assert(op1Reg != REG_NA);
+                            assert(op2Reg != REG_NA);
+
+                            if (targetReg != op1Reg)
+                            {
+                                node->SetRegNum(op1Reg);
+                                targetReg = op1Reg;
```

Comment:
We aren't doing this for any other HWIntrinsic instructions, instead we generate a `movaps` as required to put `op1Reg` in `targetReg`.

I'd expect register allocation to have already done everything at this point, but I'm not an expert here. @kunalspathak ?

Reply:
@kunalspathak Does this change looks good to you or could you give me some suggestion on  how should we handle it? Thanks.

Reply:
I agree with @tannergooding . While I have limited knowledge, but I can see that we generate `movaps` if `targetReg != op1Reg` going forward in `genHWIntrinsic_R_R_R_RM()`. Is there a situation where that doesn't happen and we have to do `targetReg = opt1Reg` explicitely?

Reply:
I double checked this part of the code and agree that we don't need to do this. I will update the code.

Reply:
While I was checking this part, I encountered some issue when I set COMPlus_jitDump=RunBasicScenario_Load. The error message I saw is as the following (same error also found in other API such as the ones in Fma_vector256):

Assert failure(PID 5076 [0x000013d4], Thread: 26664 [0x6828]): Assertion failed 'type < CORINFO_TYPE_COUNT' in 'JIT.HardwareIntrinsics.X86.SimpleTernaryOpTest__MultiplyAddDouble:RunBasicScenario_Load():this' during 'Morph - Global' (IL size 137)

    File: ...\runtime\src\coreclr\jit\ee_il_dll.hpp Line: 273
    Image: ...\runtime\artifacts\bin\coreclr\windows.x64.Debug\corerun.exe

@tannergooding, do you know what is happening here?

Reply:
I'd need to see the full stack trace, but I suspect this would be fixed by merging with (or rebasing onto) the latest dotnet/main.

Reply:
For reference, there were a few fixes that went in (https://github.com/dotnet/runtime/pull/52016, https://github.com/dotnet/runtime/pull/51731/commits/0e12823cff94fed730b213f3b3586cdd35cf7833, and https://github.com/dotnet/runtime/pull/51627) resolving asserts that had cropped up related to adding native integer support to the HWIntrinsics code.